packages/entity/src/__tests__/EntitySelfReferentialEdges-test.ts

```diff
@@ -0,0 +1,239 @@
+import Entity from '../Entity';
```

Comment:
Add cycles to this test (category A references category B which references category A). I suspect this will cause infinite loops for some deletion behaviors (namely invalidate_cache since the actual entity is not deleted).

Reply:
Nice catch, excellent suggestion. Updated the PR with code to handle and test this case.